### PR TITLE
fix(connect): device acquired handling

### PR DESCRIPTION
**Pull request by martykan**
1. Display prompt to select device when it's in an unacquired state

Screenshot 2024-02-22 at 20 16 10

This probably worked before but somehow stopped, or maybe it was intentionally changed? @mroz22 

2. Display loading after opening the popup, before device is acquired. Previously there was just a blank screen, which is bad UX in case the init takes longer. Now we keep the loading spinner before we can display something else

### CODE REVIEW

**Comment by martykan** [comment]
I solved the issue with Suite acquiring, so I think it's mergeable

File: packages/connect/src/core/index.ts [hunk 1]
```diff
@@ -231,13 +231,17 @@ const initDevice = async (method: AbstractMethod<any>) => {
 
     if (method.devicePath) {
         device = _deviceList.getDevice(method.devicePath);
-        showDeviceSelection = !device || !!device?.unreadableError;
+        showDeviceSelection =
+            !device ||
+            !!device?.unreadableError ||
+            // Suite calls getFeatures on unacquired device to acquire it
+            (device.isUnacquired() && method.name != 'getFeatures');
```
Comment (mroz22) [inline packages/connect/src/core/index.ts]: hmm, it makes sense actually but I don't like this line very much. 

the only other option would could be doing automated acquire in this case 🤔

Comment (martykan) [reply packages/connect/src/core/index.ts]: To solve this more cleanly we would probably have to add a new method/parameter that Suite can call to acquire the device
https://github.com/trezor/trezor-suite/blob/185a936bc0cf12cffa30ac4ff5c1629da44d6178/suite-common/wallet-core/src/device/deviceThunks.ts#L253-L278

**Review by mroz22 — APPROVED** [review]

File: packages/connect/src/core/index.ts (hunk 1 above)
Comment (mroz22) [reply packages/connect/src/core/index.ts]: maybe lets check if using popup

**Pull request merged**